### PR TITLE
Add wrappers around workflow models to gradually reveal postables instead of all at once

### PR DESCRIPTION
**Pull request by lukemelia**
https://www.loom.com/share/3c2dddb2777142219267027b647bf270

### CODE REVIEW

**Comment by Aierie** [comment]
Edge cases:
- Rolling back the thread when users edit a card (`onIncomplete`)
  - We need to mark messages as incomplete in our animated classes
- The initial loading of messages if the wallet connection steps are completed before starting the workflow
  - Currently this scrolls each time a message is shown and can be a bit janky if we show messages quickly.

**Comment by Aierie** [comment]
This PR does not include CSS changes. I think CSS animations are feasible and can be included in another PR.

File: packages/web-client/app/models/animated-workflow.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      model.cancelationMessages
+    );
+    this.milestones = model.milestones.map((m) => new AnimatedMilestone(m));
+    taskFor(this.tickerTask).perform();
+  }
+  @reads('model.name') declare name: string;
+
+  @task
+  *tickerTask() {
+    this.startTestWaiter();
```
Comment (lukemelia) [inline packages/web-client/app/models/animated-workflow.ts:133]: I don't understand why we are starting the waiter once, and stopping it many times.

File: packages/web-client/app/models/workflow/workflow-card.ts [hunk 2]
```diff
@@ -25,8 +26,10 @@ export class WorkflowCard extends WorkflowPostable {
 
   @action onComplete() {
     this.isComplete = true;
+    this.workflow?.animatedWrapper?.startTestWaiter();
```
Comment (lukemelia) [inline packages/web-client/app/models/workflow/workflow-card.ts:29]: This class should not know about animation or test waiters. This should be completely encapsulated in the AnimatedWorkflow. So this line should be something like `this.workflow?.onPostableComplete(this)` and then that should in turn do what needs to be done.

File: packages/web-client/app/models/workflow/workflow-card.ts [hunk 3]
```diff
@@ -25,8 +26,10 @@ export class WorkflowCard extends WorkflowPostable {
...
   }
   @action onIncomplete() {
+    this.workflow?.animatedWrapper?.startTestWaiter();
```
Comment (lukemelia) [inline packages/web-client/app/models/workflow/workflow-card.ts:32]: Same as my comment for complete handling.

File: packages/web-client/tests/acceptance/deposit-test.ts [hunk 4]
```diff
@@ -81,7 +81,8 @@ module('Acceptance | deposit', function (hooks) {
     assert.dom(`${post} [data-test-balance="DAI"]`).containsText('250.50');
     assert.dom(`${post} [data-test-balance="CARD"]`).containsText('10000.00');
 
-    await waitFor(milestoneCompletedSel(0));
```
Comment (lukemelia) [inline packages/web-client/tests/acceptance/deposit-test.ts:84]: Why would this need to change? Same for all similar test changes.

File: .vscode/settings.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
   },
   "[typescript]": {
     "editor.formatOnSave": true,
-    "editor.defaultFormatter": "esbenp.prettier-vscode"
+    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
```
Comment (lukemelia) [inline .vscode/settings.json:10]: This probably doesn't belong?

Comment (Aierie) [reply .vscode/settings.json:10]: Oops. Yep!

File: packages/web-client/app/models/animated-workflow.ts [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+    // short timeout to prevent moving revealpointer within the same runloop
+    yield rawTimeout(10);
+
+    while (true) {
+      let result = this.revealNext();
+
+      // the last card in these things is not completed so isComplete will never
+      // be true. We check for completion by making sure all the things we need to
+      // show are shown
```
Comment (lukemelia) [inline packages/web-client/app/models/animated-workflow.ts:169]: > the last card in these things is not completed
Why is this true? Does it need to be?

Comment (Aierie) [reply packages/web-client/app/models/animated-workflow.ts:169]: I think it doesn't have to be. We could hardcode the "Next steps" CTA to have `isComplete` be true.

Alternatively, we could also subclass `PostableCollection` and have it determine `isComplete` based on whether all necessary postables are visible.

Comment (lukemelia) [reply packages/web-client/app/models/animated-workflow.ts:169]: > I think it doesn't have to be. We could hardcode the "Next steps" CTA to have isComplete be true.

This! 👍

Comment (Aierie) [reply packages/web-client/app/models/animated-workflow.ts:169]: Prior to this, will be changing the way that styling is applied to workflow cards to allow cards to override default width assignment/allocation by the thread. 

After making "Next steps" completed, will replace this check with `PostableCollection.isComplete`.

These changes will be made in future PRs.

**Pull request merged**